core/amber/src/main/scala/edu/uci/ics/amber/engine/architecture/pythonworker/PythonProxyClient.scala

```diff
@@ -0,0 +1,142 @@
+package edu.uci.ics.amber.engine.architecture.pythonworker
+
+import edu.uci.ics.amber.engine.architecture.pythonworker.WorkerBatchInternalQueue.{
+  ControlElement,
+  ControlElementV2,
+  DataElement
+}
+import edu.uci.ics.amber.engine.common.ambermessage.InvocationConvertUtils.{
+  controlInvocationToV2,
+  returnInvocationToV2
+}
+import edu.uci.ics.amber.engine.common.ambermessage.{PythonControlMessage, _}
+import edu.uci.ics.amber.engine.common.rpc.AsyncRPCClient.{ControlInvocation, ReturnInvocation}
+import edu.uci.ics.amber.engine.common.virtualidentity.ActorVirtualIdentity
+import edu.uci.ics.texera.workflow.common.tuple.Tuple
+import edu.uci.ics.texera.workflow.common.tuple.schema.Schema
+import org.apache.arrow.flight._
+import org.apache.arrow.memory.{BufferAllocator, RootAllocator}
+import org.apache.arrow.vector.VectorSchemaRoot
+
+import scala.collection.mutable
+
+class PythonProxyClient(portNumber: Int)
+    extends Runnable
+    with AutoCloseable
+    with WorkerBatchInternalQueue {
+
+  val allocator: BufferAllocator =
+    new RootAllocator().newChildAllocator("flight-client", 0, Long.MaxValue)
+  val location: Location = Location.forGrpcInsecure("localhost", portNumber)
+
+  private val MAX_TRY_COUNT: Int = 3
+  private val WAIT_TIME_MS = 500
+  private var flightClient: FlightClient = _
+
+  override def run(): Unit = {
+    establishConnection()
+    mainLoop()
+  }
+
+  def establishConnection(): Unit = {
+    var connected = false
+    var tryCount = 0
+    while (!connected && tryCount < MAX_TRY_COUNT) {
+      try {
+        println("trying to connect to " + location)
+        flightClient = FlightClient.builder(allocator, location).build()
+        connected = new String(flightClient.doAction(new Action("heartbeat")).next.getBody) == "ack"
+        if (!connected) Thread.sleep(WAIT_TIME_MS)
+      } catch {
+        case _: FlightRuntimeException =>
+          println("Flight CLIENT:\tNot connected to the server in this try.")
```

Comment:
maybe better to also include the flight exception message for debugging

Reply:
the exception has a bigger issue:
when an error exception is raised, a control message `FatalError` is supposed to be sent back to Controller. Currently I disabled the rpcclient for the java side, with some reasons. Thus, it is not easy to send back exception messages to the frontend from java side now.

The change I've made based on the comment:
1. I used logger to log all those messages to console.
2. I temporarily disabled the logErrorAction which sends the error to frontend. This needs a better design to do so.